dask_expr/_merge.py

```diff
@@ -103,13 +108,44 @@ def _divisions(self):
         if self.merge_indexed_left and self.merge_indexed_right:
             return list(unique(merge_sorted(self.left.divisions, self.right.divisions)))
 
+        if self.is_broadcast_join:
+            if self.broadcast_side == "left":
+                return self.right._divisions()
+            return self.left._divisions()
+
         if self._is_single_partition_broadcast:
             _npartitions = max(self.left.npartitions, self.right.npartitions)
         else:
             _npartitions = self._npartitions
 
         return (None,) * (_npartitions + 1)
 
+    @functools.cached_property
+    def broadcast_side(self):
+        return "left" if self.left.npartitions < self.right.npartitions else "right"
+
+    @functools.cached_property
+    def is_broadcast_join(self):
+        broadcast_bias, broadcast = 0.5, None
+        broadcast_side = self.broadcast_side
+        if isinstance(self.broadcast, float):
+            broadcast_bias = self.broadcast
+        elif isinstance(self.broadcast, bool):
+            broadcast = self.broadcast
+
+        s_backend = self.shuffle_backend or get_default_shuffle_method()
+        if (
+            s_backend in ("tasks", "p2p")
```

Comment:
I assume you are just copying dask/dask logic but in the case where the shuffle backend is not explicitly set and all other conditions evaluate to true, I would expect it to be beneficial to switch from disk to broadcast

Reply:
we can't end up with disk from the ``get_default_shuffle_method``, so we will only end up with disk if it's explicitly given